src/OOPMacroImpl.jl

```diff
@@ -95,13 +95,23 @@ macro class(args...)
           end"""
 
     # this allows calling functions on the class..
-    clsFnNameList = join(map(fn->":$(getFnName(fn, withoutGeneric=true)),", collect(values(methods))),"")
+    clsFnNames = map(fn->"$(getFnName(fn, withoutGeneric=true))", collect(values(methods)))
+    clsFnNameList = join(map(name->":$name,", clsFnNames),"")
     dotAccStr = """
-        function Base.getproperty(self::$clsName, name::Symbol)
-            if isdefined(self, name) || name ∉ ($clsFnNameList)
-                getfield(self, name)
+        function Base.getproperty(self::$clsName, nameSymbol::Symbol)
+            if isdefined(self, nameSymbol) || nameSymbol ∉ ($clsFnNameList)
+                getfield(self, nameSymbol)
             else
-                (args...; kwargs...)->eval(:(\$name(\$self, \$args...; \$kwargs...)))
+                if haskey($(OOPMacroModule).clsFnDefDict, self)
```

Comment:
Instead of caching the functions on the fly inside getproperty, try cache them outside

Reply:
I'm not sure I understand. Maybe we can initialize it somehow in the constructor.